bestla/bestla_prologue_b.h

```diff
@@ -731,6 +795,22 @@ class WeightKBlockNInteger {
         } else {
           assert(0);
         }
+      } else if (wptr->SDtype() == BTLA_DTYPE::DQ8_BNB) {
+        auto internal_n_offset = n_offset + i;
+        if (wptr->mDType == BTLA_DTYPE::S4_CLIP) {
+          kernel::wrapper::DecompressDQKBlockS4Fp<_T, _GemmCore_T::PACK_ROW>::template forward<ISA_T,
+                                                                                               BTLA_DTYPE::S4_CLIP>(
+              wptr->template WPtr<utils::int4x2>() + n_offset * KPad / 2 + k_offset * _GemmCore_T::NTILE / 2 +
+                  i * KPad / 2,
+              *dstptr + i * k_size, k_size / _GemmCore_T::PACK_ROW, ColSize, ColSize, ColSize,
+              wptr->template SPtr<uint8_t>(), wptr->template DQPtr<float>(), k_offset / _GemmCore_T::PACK_ROW,
+              internal_n_offset, wptr->mBlockSize / _GemmCore_T::PACK_ROW, NPad, wptr->mN, wptr->mDqBlockSize,
+              wptr->mCorrection.mDQCorrectionBuf.mBufSize / sizeof(float) - 1, tmpcache, cachesize);
```

Comment:
shall we decouple scales and weight here?

instead of w_s4_scale_s8 => s_fp in one function,
w_s4_scale_s8 => w_s4_scale_fp => w_fp with 2 functions?

Reply:
Ok this is a temporary solution, in the near feature

```getScale``` then ```getWeight```